Add Thwaites 4km test group to landice core

matthewhoffman opened the pull request:
This merge adds the Thwaites variable resolution 4-14km test group to the landice core.  This test group uses a pre-existing mesh and initial condition file and provides decomposition and restart tests.  The test structure follows closely to that used for the Greenland FO tests.  The tesst are added to the fo and full integration suites, and the test group is briefly described in the docs.  

In the future additional Thwaites tests cases could be added that actually build the mesh and initial condition.

matthewhoffman commented:
The new tests pass on Badger and are both short enough to include in the standard regression suites:
```
00:38 PASS landice_thwaites_decomposition_test
00:57 PASS landice_thwaites_restart_test
```

xylar commented on compass/landice/tests/thwaites/restart_test/__init__.py:
```diff
@@ -0,0 +1,78 @@
+        step = RunModel(test_case=self, name=name, subdir=name, cores=cores,
+                        min_cores=min_cores, threads=1)
+        # modify the namelist options and streams file
+        step.add_namelist_file(
+            'compass.landice.tests.greenland.restart_test',
+            'namelist.full', out_name='namelist.landice')
+        step.add_streams_file(
+            'compass.landice.tests.greenland.restart_test',
+            'streams.full', out_name='streams.landice')
```
```suggestion
        step.add_namelist_file(
            'compass.landice.tests.thwaites.restart_test',
            'namelist.full', out_name='namelist.landice')
        step.add_streams_file(
            'compass.landice.tests.thwaites.restart_test',
            'streams.full', out_name='streams.landice')
```

xylar replied:
If this is on purpose, it would be better to have these files in some shared (framework) space so it's clear that they're shared between test cases.  But I suspect it's just a typo.

xylar commented on compass/landice/tests/thwaites/decomposition_test/__init__.py:
```diff
@@ -0,0 +1,65 @@
+            The test group that this test case belongs to
+
+        """
+        name = 'decomposition_test'
+        subdir = '{}'.format(name)
+        super().__init__(test_group=test_group, name=name, subdir=subdir)
```
This is fine and probably made sense copying the `greenland` test cases.  Since `subdir` defaults to being the same as `name`, you can just do:
```suggestion
        super().__init__(test_group=test_group, name='decomposition_test')
```
If you like the clarity of giving the `subdir` explicitly, I would at least take out the unnecessary string formatting:
```suggestion
        name = 'decomposition_test'
        subdir = name
        super().__init__(test_group=test_group, name=name, subdir=subdir)
```

xylar commented on compass/landice/tests/thwaites/restart_test/__init__.py:
```diff
@@ -0,0 +1,78 @@
+            The test group that this test case belongs to
+
+        """
+        name = 'restart_test'
+        subdir = '{}'.format(name)
+        super().__init__(test_group=test_group, name=name, subdir=subdir)
```
Same comments as for the `DecompositionTest`.

xylar commented on compass/landice/tests/thwaites/run_model.py:
```diff
@@ -0,0 +1,96 @@
+
+        suffixes : list of str, optional
+            a list of suffixes for namelist and streams files produced
+            for this step.  Most steps most runs will just have a
+            ``namelist.landice`` and a ``streams.landice`` (the default) but
+            the ``restart_run`` step of the ``restart_test`` runs the model
+            twice, the second time with ``namelist.landice.rst`` and
+            ``streams.landice.rst``
```
```suggestion
            for this step.  Most run steps will just have a
            ``namelist.landice`` and a ``streams.landice`` (the default) but
            the ``restart_run`` step of the ``restart_test`` runs the model
            twice, the second time with ``namelist.landice.rst`` and
            ``streams.landice.rst``
```
I'm guessing this typo is in all `RunModel` steps in `landice`. Sorry about that!

xylar commented on docs/users_guide/landice/test_groups/index.rst:
```diff
@@ -5,8 +5,8 @@ Test Groups
-:ref:`landice_suite_sia_integration`.  We expect to port many more test cases
-in the coming months.
+:ref:`landice_suite_sia_integration` and many of the test cases in the
+:ref:`landice_suite_fo_integration`.
```
This label needs to be added to `docs/users_guide/landice/suites.rst`:

```
...
(``restart_test``). Others ensure that the code produces the same result when
it is run on different numbers of cores (``decomposition_test``).

.. _landice_suite_fo_integration:

fo_integration test suite
-------------------------
...
```

xylar commented:
I noticed a few things in my review.  The only one directly related to the PR is that there's not yet a section for the `thwaites` test group in the developer's guide.

Not directly related to this PR:
* the mesh link in the `greenland` test group (both in the user's and developer's guides) seems to be the old one (date stamp `150922`).
* You didn't make any changes to the developer's guide for `greenland` in #146, and that probably should also have been updated to reflect the new FO test cases.

Sorry I missed those 2.

matthewhoffman commented:
@xylar , thanks for looking this over and providing me feedback.  I've fixed all the items and updated the missing parts of the docs.  Feel free to look it over again if you'd like, but you don't need to.

xylar commented on docs/users_guide/landice/test_groups/index.rst:
```diff
@@ -15,4 +15,5 @@ in the coming months.
    enthalpy_benchmark
    eismint2
    greenland
+   thwaites
```
The corresponding addition is needed in the `docs/developers_guide/landice/test_groups/index.rst`.

Also, so far the tests have been alphabetical.  There's no reason that needs to be the case.  You may want to give them a more logical order if there is one.  That could be something for another PR, perhaps as more test groups get added

xylar commented on docs/developers_guide/landice/test_groups/thwaites.rst:
```diff
@@ -0,0 +1,64 @@
+thwaites
+=========
+
+The ``thwaites`` test group (:py:class:`compass.landice.tests.thwaites.thwaites`)
```
```suggestion
The ``thwaites`` test group (:py:class:`compass.landice.tests.thwaites.Thwaites`)
```

xylar commented:
Two more small things.
